src/targets/__tests__/maven.test.ts

```diff
@@ -351,3 +353,24 @@ describe('get gradle home directory', () => {
     expect(actual).toEqual(expected);
   });
 });
+
+describe('createUserGradlePropsFile', () => {
+  const gradleHomeEnvVar = 'GRADLE_USER_HOME';
+
+  afterEach(() => {
+    delete process.env[gradleHomeEnvVar];
+  });
+
+  test('should make sure that directory exists before writing gradle props file', async () => {
+    const randomTmpDir = '/random/depth/of/directories';
+    process.env[gradleHomeEnvVar] = randomTmpDir;
+    await createMavenTarget().createUserGradlePropsFile();
+    expect(fsPromises.mkdir).toHaveBeenCalledWith(randomTmpDir, {
+      recursive: true,
+    });
```

Comment:
Considering there's a helper [`withTempDir`](https://github.com/getsentry/craft/blob/master/src/utils/files.ts#L76) function, I think it'd be better to not be that implementation-specific and abstract how the method is behaving. With an actual temporary directory, test whether the gradle props file exists after the method has been executed (and two use cases here: when the gradle directory exists, and when it doesn't). This approach doesn't rely on implementation details.

Reply:
Agree, although its not that trivial to do with all the mocking done in this file. I can improve on those once we get this thing working in the first place. Appreciate the after-closing review!